Source/Tests/LinuxSyscalls/Syscalls.cpp

```diff
@@ -226,89 +211,156 @@ static bool IsSupportedByInterpreter(std::string const &Filename) {
   return false;
 }
 
-uint64_t ExecveHandler(const char *pathname, char* const* argv, char* const* envp, ExecveAtArgs *Args) {
+static bool IsShebangFilename(std::string const &Filename) {
+  // Open the Filename to determine if it is a shebang file.
+  std::fstream File;
+  size_t FileSize{0};
+  File.open(Filename, std::fstream::in | std::fstream::binary);
+
+  if (!File.is_open())
+    return false;
+
+  File.seekg(0, File.end);
+  FileSize = File.tellg();
+  File.seekg(0, File.beg);
+
+  // Is the file large enough for shebang
+  if (FileSize <= 2)
+    return false;
+
+  // The maximum length of the shebang line is `#!` + 255 chars
+  std::vector<char> Header(std::min(257ul, FileSize));
+  File.read(&Header.at(0), Header.size());
+
+  return IsShebangFile(&Header.at(0), Header.size());
+}
+
+static bool IsShebangFD(int FD) {
+  // We don't know the state of the FD coming in since this might be a guest tracked FD.
+  // Need to be extra careful here not to adjust file offsets and status flags.
+  //
+  // Can't use dup since that makes the FD have the same file description backing both FDs.
+
+  // We need to first determine the file size through fstat.
+  struct stat buf{};
+  if (fstat(FD, &buf) == -1) {
+    // Couldn't get size.
+    return false;
+  }
+
+  // Is the file large enough for shebang
+  if (buf.st_size <= 2)
+    return false;
+
+  // The maximum length of the shebang line is `#!` + 255 chars
+  std::vector<char> Header(std::min(257l, buf.st_size));
```

Comment:
=> std::array

Reply:
Some of the logic with this comment and others conflicted. So you'll need to check that again with the changes.